_posts/2020-06-18-a-sustainable-nuget-marketplace-will-have-to-compete-with-the-nuget-gallery.markdown

```diff
@@ -0,0 +1,368 @@
+
+So what can solve the sustainability issue?
+
+## Another solution: a NuGet marketplace
+
+Looking at foundations and sponsorships, there are a few things that make these solutions suboptimal for producers
+and consumers.
+
+Producers will have to make consumers aware that there are paid options. They have to direct them to the sponsor button,
+and can say the open source foundation lawyers approve of the license.
+
+As a consumer, you have to be aware of these options, and actively search for them. They are not "in your face"
+in our IDE, where we consume most open source through the NuGet client. Which means consumers often have no idea
+there is a sponsor button.
+
+This is where https://sdkbin.com, and https://bytepack.io, and others come in. In fact, our community has been
+dreaming this up for a long, long time.
+
+Some recent calls for this:
+
+<blockquote class="twitter-tweet"><p lang="en" dir="ltr">An idea.<br><br>A managed NuGet[org] alternative for selling your NuGet packages.<br><br>A web interface for discovering paid-for packages. After purchase, they are made available (forever) in a private feed.<br><br>Subscriptions and payment are managed by the platform.</p>&mdash; Paul Knopf (@pauldotknopf) <a href="https://twitter.com/pauldotknopf/status/1179129145235820544?ref_src=twsrc%5Etfw">October 1, 2019</a></blockquote> <script async src="https://platform.twitter.com/widgets.js" charset="utf-8"></script>
+
+<blockquote class="twitter-tweet"><p lang="en" dir="ltr">We basically need to transform our NuGet, Npm, VSCode plugin marketplace whatever into AppStore. <br><br>If dual licensing is not supported out of the box it will never become mainstream solution for sustaining OSS <a href="https://t.co/oRllcH0qYA">https://t.co/oRllcH0qYA</a></p>&mdash; Krzysztof Cieślak @ #BlackLivesMatter (@k_cieslak) <a href="https://twitter.com/k_cieslak/status/1222217476051742720?ref_src=twsrc%5Etfw">January 28, 2020</a></blockquote> <script async src="https://platform.twitter.com/widgets.js" charset="utf-8"></script>
+
+I know when I was building [MyGet](https://www.myget.org/) with [Xavier](https://twitter.com/xavierdecoster),
+we dreamt up something like a NuGet marketplace as well. Will have to check but I think I still have `nugetmarketplace.com`
+somewhere...
+
+The above is scientific proof a NuGet marketplace is needed!
+Let's look at what a NuGet marketplace should help with.
+
+### For producers: legal services
+
+Open source projects can still be open source, but sell additional services. These could be:
+
+* Dual-licensed open source
+* Paid add-on packages
+* Support services
+* Consulting services
+
+Dual-licensing will require a set of templates that projects can work with.
+If the project uses Apache-2, MIT, GPL, or other open source licenses, what will a commercial dual-license
+look like?
+
+Paid add-on packages may be open or closed source, but they will still need a license. A NuGet marketplace
```

Comment:
Yep. Must have standard commercial licenses - because _the people buying them_ will also need to trust that those license agreements won't come back to haunt them down the road. One of the big incentives for adopting OSS right now is the fact that most licenses are permissive - a hurdle we're going to have to overcome is making less permissive, proprietary licenses more palatable.

Reply:
Can I take that as a quote here?

Reply:
Go for it